src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -893,7 +891,7 @@ private ZipArchiveEntry CreateEntry(ZipArchive package, string entryName, Compre
         private ZipArchiveEntry CreatePackageFileEntry(ZipArchive package, string entryName, DateTimeOffset timeOffset, CompressionLevel compressionLevel)
         {
             var entry = package.CreateEntry(entryName, compressionLevel);
-            entry.LastWriteTime = timeOffset;
+            entry.LastWriteTime = timeOffset < ZipFormatMinDate ? ZipFormatMinDate : timeOffset;
```

Comment:
Timezones! 

The set date should be ZipFormatMinDate + a few days. This would mean that re-packaged files would not hit this issue again.

Reply:
Should we add a warning? 
Maybe not, but I think at the minimum we should add a higher verbosity message about the changes we might be making.

Reply:
@nkolev92 
Can you give me code link for how to show "higher verbosity message"? Currently I don't know how to do that, only thing I know is throwing exception always end up on somewhere on UI.

Reply:
logger...

Reply:
@nkolev92 
Can you tell me which logger? It looks there more than 1 logger exist, but non in this file.

Reply:
@nkolev92 
I noticed 2 problems with using logger with logger to inform about lastwritetime(modified) timestamp was change.
1. There are some packages have so many times like some content files (specially web related) etc.
Then log will include so many lines of below info, I not sure if it's good experience for user.

![image](https://user-images.githubusercontent.com/8766776/101970397-f5599780-3bde-11eb-8835-b3da67f2218a.png)

2. Also we're already have existing code does similar thing but it doesn't have any logging. I'm not sure what is this `_deterministic` flag, but now it's always disabled by default on [line#30](https://github.com/NuGet/NuGet.Client/blob/aa3d157d51cb97e0319644a7e54ad90d6eea3457/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs#L96). Do we need to do same thing here or not?

https://github.com/NuGet/NuGet.Client/blob/aa3d157d51cb97e0319644a7e54ad90d6eea3457/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs#L886-L889

https://github.com/NuGet/NuGet.Client/blob/aa3d157d51cb97e0319644a7e54ad90d6eea3457/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs#L926-L930

@aortiz-msft  @JonDouglas can also give your input for wording. 
Currently I have something like this `bin/Debug/netcoreapp3.1/NuGetDateTimeOffsetIssue1.deps.json file lastwritetime(modified) timestamp changed from 12/10/1977 to 1/4/1980 due to zip file doesn't support timestamp before 01/01/1980 limitation.`

Reply:
`{0} file lastwritetime(modified) timestamp changed from {1} to {2} due to zip file doesn't support timestamp before 01/01/1980 limitation.`

->

Do we have more documentation on this limitation? I believe this is a "strict timestamp" for 1980-01-01, but this also happens for 2107-12-31 correct?

@zkat I believe you may have experience with this based on https://github.com/npm/npm/issues/19968 any further ideas on messaging here?

Reply:
@JonDouglas 
It's documented on [ZipArchiveEntry.xml](https://github.com/dotnet/dotnet-api-docs/blob/0cdc9c94d4b5eccf9d6b6bae7565401529de7aa1/xml/System.IO.Compression/ZipArchiveEntry.xml#L336), not sure if we have any documentation on our side.
`The earliest permitted value is 1980 January 1 0:00:00 (midnight). The latest permitted value is 2107 December 31 23:59:58`

For practicality I guess we shouldn't worry about files after 2107 for now. There're more close practical problems before 2107 like [Year2038](https://en.wikipedia.org/wiki/Year_2038_problem) problem.

Reply:
> The set date should be ZipFormatMinDate + a few days.

I disagree. We should internally use UTC time, and then timezone doesn't matter. Let the operating system figure out the filesystem's timezone info. And using + a few days makes me feel like we don't understand how to do timezones properly, so using a hack instead of learning how to do it.  I think we can do better.

> {0} file lastwritetime(modified) timestamp changed from {1} to {2} due to zip file doesn't support timestamp before 01/01/1980 limitation.

My suggestion is to use a message something like: `Clipping the timestamp of {0}, as {1} does not fit in the ZIP format `.  Does ZIP allow different creation date from modified date?  If not, having `lastwritetime(modified)`, or something similar, is meaningless. Even if zip does allow for multiple different types of timestamps, is it meaningful to the customer telling them which timestamp is being clipped into the allowed range?

Reply:
> I disagree. We should internally use UTC time, and then timezone doesn't matter. 

I don't deny that, but the tail is likely longer :) In the grand scheme, it doesn't matter too much what date we pick I think. 

Now might be a good time to clean-up our date usage though.

Reply:
Even if it's a lot of work to change all our usage of date/times in our code to UTC, we can still use [`DateTimeOffset.ToUniversalTime`](https://docs.microsoft.com/en-us/dotnet/api/system.datetimeoffset.touniversaltime?view=net-5.0) to convert it to UTC in this method. Adding extra days to avoid timezone issues feels like using a bigger hammer, rather than a better tool.

Reply:
Ok. Changed to UTC version and also added max date support.